### DISCUSSION TITLE
Questions about XFEM Module input file and crack propagation criterion

### DISCUSSION THREAD

**Discussion by emdcstanford** [idaholab/moose#17970] [Q&A Modules: General] [answered]
Hi,

I am fairly new to using the XFEM module in MOOSE, and I've been running some of the test simulations as well as some of my own. However, I'm a little confused about some of the parameters in the example input files, especially in relation to the  crack propagation and initiation theory presented in the paper [Jiang et al. 2020 in Eng. Frac. Mech](https://www.sciencedirect.com/science/article/abs/pii/S0013794419307568).

With regards to the 'crack_growth_increment parameter', is this the amount of distance the crack is prescribed to grow each time the crack propagation criterion is satisfied? However I notice I can also propagate the crack without this parameter.

Then in XFEMRankTwoTensorMarkerUserObject, I am a little confused about the 'threshold'. Is this a scalar quantity equivalent to the fracture strength described in the paper mentioned above for crack initiation (section 5.2) which we use when there are no prescribed cracks apriori, or is this part of the crack propagation criterion? I was under the impression that the propagation criterion is that of maximum hoop stress and don't immediately see how this is related to a threshold value.

Thank you very much for the help,

Enrique

**Comment by jiangwen84** [comment, marked as answer]
@emdcstanford Some of the crack propagation criterions mentioned in that paper has not been fully merged into the MOOSE repository. It definitely reminds us to get those codes merged as soon as we can. 

You are correct that the crack propagation criterion was based on maximum hoop stress in that paper. And the maximum hoop stress is obtained by evaluating the stress intensity factors. In MOOSE, `XFEMRankTwoTensorMarkerUserObject` uses a different simple criterion where the crack propagates if the crack tip element stress exceeds a threshold and the direction is determined as the maximum principal direction. 

Regarding the `crack_growth_increment`, this parameter specifies the amount of distance the crack will grow when crack propagation criterion is met. Without providing this parameter, crack will only propagate one element per time.

Let us know if you have more questions.
